### PR TITLE
Fix issue #1753 Reject promise when workbook reader is writing to temporary file stream and error occurs

### PR DESCRIPTION
**Pull request by pauliusg**
## Summary

Fixes issue https://github.com/exceljs/exceljs/issues/1753

Workbook code which is creating temporary file by using write stream is not watching for error event, so in case of error promise was not resolved and code stays in hanging state.

### CODE REVIEW

File: lib/stream/xlsx/workbook-reader.js
Comment (pauliusg) [inline lib/stream/xlsx/workbook-reader.js:302]: This only formatting applied by pre-commit hook.

Comment (pauliusg) [inline lib/stream/xlsx/workbook-reader.js:302]: This only formatting applied by pre-commit hook.

**Comment by pauliusg** [comment]
Why this one is still not merged?

**Comment by Siemienik** [comment]
Hello hello @pauliusg , for quite a long time none of the maintainers had enough time to properly take care of this project. Especially since it is being created voluntarily in their free time. Thank you very much for your contribution, and I will try to merge this PR as soon as possible. 
By the way, yesterday we did our first "MergeFest" and there were 3 of us on the online stream. Today we plan to merge more PRs. It would be great if you could join us, you could guide us through this PR and the review would be much easier :)
[Link to the event on DISCORD](https://discord.gg/J9PmAjup?event=1096113161750454313
), on time: 3PM CEST
I also invite you to join the community on our server https://discord.gg/SNEtBKPUxE

Have a nice day! :)

**Comment by pauliusg** [comment]
@Siemienik thanks for response. It is good to know that project is still alive! Unfortunately I can't join you, but this PR very simple one-liner to properly handle stream errors. More context is in issue: https://github.com/exceljs/exceljs/issues/1753

**Review by Siemienik — APPROVED** [review]
I really appreciate your contribution, thank you :) 
It looks quite obvious and will merge it asap after the test will be added,

**Review by naorpeled — APPROVED** [review]

**Review by coder-xiaotian — COMMENTED** [review]
LGTM

**Pull request merged**
